### PR TITLE
dune: Add default alias = install alias

### PR DESCRIPTION
**Pull request by SkySkimmer**
The choice of contents may be debated.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
PS this seems to be broken:
~~~
$ dune build
File "doc/dune", line 68, characters 9-20:
68 |  (files (refman-html as html/refman) (refman-pdf as pdf/refman))
              ^^^^^^^^^^^
Error: Rule failed to generate the following targets:
- default/doc/coq-doc/html/refman
File "doc/dune", line 68, characters 38-48:
68 |  (files (refman-html as html/refman) (refman-pdf as pdf/refman))
                                           ^^^^^^^^^^
Error: Rule failed to generate the following targets:
- default/doc/coq-doc/pdf/refman
File "doc/stdlib/dune", line 43, characters 9-13:
43 |  (files (html as html/stdlib))
              ^^^^
Error: Rule failed to generate the following targets:
- default/doc/coq-doc/html/stdlib
~~~

**Comment by ejgallego** [comment]
> PS this seems to be broken:

This seems indeed like the transitive aliases dune bug, which has been fortunately fixed in Dune's `main` but it was indeed tricky.

**Comment by ejgallego** [comment]
> This seems indeed like the transitive aliases dune bug, which has been fortunately fixed in Dune's `main` but it was indeed tricky.

As a workaround, add the missing doc alias to `default`, that may fix it [we have other pitfalls arising from this bug but we will have to wait for 3.0]

File: dune [hunk 1]
```diff
@@ -18,6 +18,10 @@
 ;
 ; (_ (flags :standard -rectypes)))
 
+(alias
+ (name default)
+ (deps (alias install)))
```
Comment (Zimmi48) [inline dune:23]: I would be more radical and make the default alias equate the `world` target, i.e. only `coq-core.install coq-stdlib.install coqide-server.install coqide.install` (non-doc install targets).

**Comment by SkySkimmer** [comment]
>This seems indeed like the transitive aliases dune bug, which has been fortunately fixed in Dune's main but it was indeed tricky.

I don't think so, `dune build @install` and `dune build coq-doc.install` fail the same way, and I have the refman files in _build/default.

**Comment by ejgallego** [comment]
> I don't think so, `dune build @install` and `dune build coq-doc.install` fail the same way, and I have the refman files in _build/default.

Umm, indeed it seems the hack we were using to install dirs is not working anymore :( , see the comment in `doc/dune` , I dunno what we should do.

**Comment by Zimmi48** [comment]
Did it ever work? I don't recall so.

**Comment by ejgallego** [comment]
> Did it ever work? I don't recall so.

I don't remember, so indeed it is possible it never worked. Indeed opam install files would require listing all files, not a single directory, so I dunno how to approach this.

**Comment by Zimmi48** [comment]
For this PR, the easiest solution is simply to not include `coq-doc` in the default alias.

As for how to fix `coq-doc.install`, maybe the coq-doc package should rely on `make` rather than `dune` to build.

**Comment by ejgallego** [comment]
> As for how to fix `coq-doc.install`, maybe the coq-doc package should rely on `make` rather than `dune` to build.

Indeed, I think that this should be easy to do, modulo generating the dreaded `coq-doc.install` file.

**Comment by SkySkimmer** [comment]
Changed to not include doc

**Review by ejgallego — APPROVED** [review]
LGTM, @Zimmi48 ?

**Comment by Zimmi48** [comment]
LGTM

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**